climt/_components/bucket_surface/component.py

```diff
@@ -0,0 +1,197 @@
+            'units': 'kg m^-3',
+        },
+        'soil_layer_thickness': {
+            'dims': ['*'],
+            'units': 'm',
+        },
+        'heat_capacity_of_soil': {
+            'dims': ['*'],
+            'units': 'J kg^-1 degK^-1',
+        },
+        'lwe_thickness_of_soil_moisture_content':{
+            'dims': ['*'],
+            'units': 'm',
+        },
+        'convective_precipitation_rate': {
+            'dims': ['*'],
+            'units': 'm s^-1 ',
+        },
+        'stratiform_precipitation_rate': {
+            'dims': ['*'],
+            'units': 'm s^-1 ',
+        },
+        'specific_humidity': {
+            'dims': ['mid_levels', '*'],
+            'units': 'kg/kg ',
+        },
+        'surface_specific_humidity': {
+            'dims': ['*'],
+            'units': 'kg/kg ',
+        },
+        'air_temperature': {
+            'dims': ['mid_levels', '*'],
+            'units': 'degK ',
+        },
+        'northward_wind': {
+            'dims': ['mid_levels', '*'],
+            'units': 'm s^-1',
+        },
+        'eastward_wind': {
+            'dims': ['mid_levels', '*'],
+            'units': 'm s^-1',
+        },
+    }
+
+    diagnostic_properties = {
+        'precipitation_rate': {
+            'dims': ['*'],
+            'units': 'm s^-1 ',
+        },
+        'surface_upward_latent_heat_flux': {
+            'dims': ['*'],
+            'units': 'W m^-2',
+        },
+        'surface_upward_sensible_heat_flux': {
+            'dims': ['*'],
+            'units': 'W m^-2',
+        },
+        'evaporation_rate': {
+            'dims': ['*'],
+            'units': 'm s^-1',
+        },
+    }
+
+    output_properties = {
+        'surface_temperature': {'units': 'degK'},
+        'lwe_thickness_of_soil_moisture_content':{'units': 'm'},
+    }
+
+
+    def __init__(self, soil_moisture_max=0.15, g=0.75, specific_latent_heat_of_water=2260000,
+                 bulk_coefficient=0.0011, **kwargs):
+        """
+        Args:
+
+        soil_moisture_max:
+            The maximum moisture that can be held by the surface_temperature
+        g:
+            A constant value that is used in the beta_factor calculation.
+        bulk_coefficient:
+            The bulk transfer coeffiecient that is used to calculate
+            maximum evaporation rate and sensible heat flux
+        """
+        self._smax = soil_moisture_max
+        self._g = g
+        self._c = bulk_coefficient
+        self._l = specific_latent_heat_of_water
+        super(BucketSurface, self).__init__(**kwargs)
+
+
+    def array_call(self, state, timestep):
+        '''
+        Calculates sensible and latent heat flux and returns
+        surface temperature and soil moisture after timestep.
+        '''
+
+        beta_factor = 0
+
+        wind_speed = sqrt(pow(state['northward_wind'][0], 2) + \
+           pow(state['eastward_wind'][0], 2))
+        evaporation_rate_max = self._c * wind_speed * \
```

Comment:
rename `evaporation_rate_max` to `potential_evaporation` (again a technically used term).